pkg/csi/service/wcpguest/controller.go

```diff
@@ -420,6 +453,128 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
+			if key == common.Nfsv4AccessPointKey {
+				publishInfo[common.Nfsv4AccessPoint] = value
+				break
+			}
+		}
+		resp := &csi.ControllerPublishVolumeResponse{
+			PublishContext: publishInfo,
+		}
+		log.Infof("ControllerPublishVolume: Volume %q attached successfully on the node: %q", req.VolumeId, req.NodeId)
+		return resp, nil
+	}
+	cnsFileAccessConfigWatcher, err := k8s.NewCnsFileAccessConfigWatcher(ctx, c.restClientConfig, c.supervisorNamespace)
+	if err != nil {
+		msg := fmt.Sprintf("failed to create cnsFileAccessConfigWatcher. Error: %+v", err)
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	// Attacher timeout, default is set to 4 minutes
+	timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
+	// Adding watch on the CnsFileAccessConfig instance to register for updates
+	watchCnsFileAccessConfig, err := cnsFileAccessConfigWatcher.Watch(metav1.ListOptions{
+		FieldSelector:   fields.SelectorFromSet(fields.Set{"metadata.name": cnsFileAccessConfigInstance.Name}).String(),
+		ResourceVersion: cnsFileAccessConfigInstance.ResourceVersion,
+		TimeoutSeconds:  &timeoutSeconds,
+	})
+	if err != nil {
+		msg := fmt.Sprintf("failed to watch cnsfileaccessconfig %q with Error: %v", cnsFileAccessConfigInstance.Name, err)
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	defer watchCnsFileAccessConfig.Stop()
+	timeout := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
+	// Watch all update events made on CnsFileAccessConfig instance until accessPoints is set
+	for {
+		log.Debugf("Waiting for update on cnsfileaccessconfigs: %q", cnsFileAccessConfigInstance.Name)
+		event := <-watchCnsFileAccessConfig.ResultChan()
+		cnsfileaccessconfig, ok := event.Object.(*cnsfileaccessconfigv1alpha1.CnsFileAccessConfig)
+		if !ok {
+			msg := fmt.Sprintf("Watch on cnsfileaccessconfig instance %q timed out", cnsFileAccessConfigInstance.Name)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		// Check if SV PVC Name and VM name in the cnsfileaccessconfig instance match with VolumeId and NodeId from the request
+		if cnsfileaccessconfig.Spec.PvcName != req.VolumeId && cnsfileaccessconfig.Spec.VMName != req.NodeId {
+			log.Debugf("Observed vm name: %q, expecting vm name: %q. Observed SV PVC Name: %q, expecting SV PVC Name: %q. Continuing...", cnsfileaccessconfig.Spec.VMName, req.NodeId, cnsfileaccessconfig.Spec.PvcName, req.VolumeId)
+			continue
+		}
+		log.Debugf("Observed an update on cnsfileaccessconfig: %+v", cnsfileaccessconfig)
+		if cnsfileaccessconfig.Status.Done && cnsfileaccessconfig.Status.Error == "" {
+			publishInfo := make(map[string]string)
+			// Check if the updated instance has the AccessPoints
+			for key, value := range cnsfileaccessconfig.Status.AccessPoints {
+				if key == common.Nfsv4AccessPointKey {
+					publishInfo[common.Nfsv4AccessPoint] = value
+					break
+				}
+			}
+			return &csi.ControllerPublishVolumeResponse{
```

Comment:
Same. Set the disk type as file volume in publishInfo before returning response.